python_configure.bzl

```diff
@@ -167,6 +168,23 @@ def _get_python_bin(repository_ctx):
             python_bin = _norm_path(python_bin)
         return python_bin
 
+    if (
```

Comment:
The way this is trying to work is fundamentally flawed. The core issue is the repository phase can't know what interpreter a target will use, and it can't know the full set of toolchains registered. The `@pythons_hub` only keeps track of `python.toolchain()` calls, but that doesn't include any custom user toolchains.

I'm not really sure this will work, either. IIRC, the label isn't a canonical one (one that starts with '@@`, so it won't be able to resolve (you would have to use_repo() the apparent name)

From what I remember, these labels are also host specific. This is because they're just used to drive `pip.parse()` (which does various repository phase things, so has to have a host-capable interpreter available).

What you really want is to get this information from the toolchain directly. e.g., in the regular rule code, do something like

```
def _pybindthing_impl(ctx):
   py_lib_path = ctx.toolchains["pything"].lib_path
   ctx.actions.run(...stuff using py_lib_path...)
```

This should be pretty easy. If you can give a run down of the information needed, we can figure out how to make it available from the toolchains.

Reply:
Oh snap. I just realised that I still owe you a reply on https://github.com/bazelbuild/rules_python/issues/824#issuecomment-1749724164. :(